pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,259 @@
+	// Return handle to caller
+	handleChannel <- ticker
+
+	wdName := agentName + "currentProfile"
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(25 * time.Second)
+	ctx.ps.StillRunning(wdName, warningTime, errorTime)
+	ctx.ps.RegisterFileWatchdog(wdName)
+
+	for {
+		select {
+		case <-ticker.C:
+			localProfileServer = getconfigCtx.localProfileServer
+			if localProfileServer != "" {
+				start := time.Now()
+				iteration++
+				localProfileURL, err := getLocalProfileURL(localProfileServer)
+				if err != nil {
+					log.Errorf("getLocalProfileURL: %s", err)
+				} else {
+					if err := getLocalProfileConfig(localProfileURL, iteration, getconfigCtx); err != nil {
+						log.Errorf("localProfileTimerTask getLocalProfileConfig: %s", err)
+					} else {
+						publishZedAgentStatus(getconfigCtx)
+					}
+				}
+				ctx.ps.CheckMaxTimeTopic(wdName, "getLocalProfileConfig", start,
+					warningTime, errorTime)
+			}
+
+		case <-stillRunning.C:
+		}
+		ctx.ps.StillRunning(wdName, warningTime, errorTime)
+	}
+}
+
+func validateAndSetLocalProfile(localProfileBytes []byte, getconfigCtx *getconfigContext) error {
+	var localProfile = &profile.LocalProfile{}
+	err := proto.Unmarshal(localProfileBytes, localProfile)
+	if err != nil {
+		return fmt.Errorf("validateAndSetLocalProfile: Unmarshalling failed: %v", err)
+	}
+	if localProfile.GetServerToken() != getconfigCtx.profileServerToken {
+		// send something to ledmanager ??
+		return fmt.Errorf("validateAndSetLocalProfile: missamtch ServerToken for loacl profile server")
+	}
+	if getconfigCtx.currentProfile != localProfile.GetLocalProfile() {
+		log.Noticef("validateAndSetLocalProfile: profile changed from %s to %s",
+			getconfigCtx.currentProfile, localProfile.GetLocalProfile())
+		getconfigCtx.currentProfile = localProfile.GetLocalProfile()
+		writeReceivedProtoMessageLocalProfile(localProfileBytes)
+		publishZedAgentStatus(getconfigCtx)
+	}
+	return nil
+}
+
+// read saved local profile in case of particular reboot reason
+// if it is success will not read again on the next call
+func readSavedLocalProfile(getconfigCtx *getconfigContext) error {
+	localProfile, err := readSavedProtoMessage(
+		getconfigCtx.zedagentCtx.globalConfig.GlobalValueInt(types.StaleConfigTime),
+		filepath.Join(checkpointDirname, savedLocalProfileFile), false)
+	if err != nil {
+		return fmt.Errorf("lastlocalprofile: %v", err)
+	}
+	if localProfile != nil {
+		log.Function("Using saved local profile")
+		getconfigCtx.readSavedLocalProfile = true
+		return validateAndSetLocalProfile(localProfile, getconfigCtx)
+	}
+	return nil
+}
+
+//prepareLocalProfileServerMap process configuration of network instances to find match with defined localServerURL
+//returns the srcIP for the zero or more network instances on which the locaclProfileServer might be hosted
+//based on a subnet inclusion match or hostname in dns records
+//in form actual link (with hostname replaced) -> source IP
```

Comment:
Reading the code the key to the map is a URL and not a link/ifname. I think the code is wrong and should use the BridgeName in the NetworkInstanceStatus instead of some URL.

Reply:
I process provided localServerURL here to replace any HostNames inside it with IP address of AppInstance with matched HostName.

Reply:
The link name should be a string like "bn2".